src/beagle-tester.c

```diff
@@ -280,2731 +199,9 @@ static void do_fill_screen(struct fb_info *fb_info, int pattern)
-
-	ASSERT(ptr != MAP_FAILED);
-
-	fb_info->ptr = ptr;
-}
-
-static void fb_clear_area(struct fb_info *fb_info, int x, int y, int w, int h)
-{
-	int i = 0;
-	int loc;
-	char *fbuffer = (char *)fb_info->ptr;
-	struct fb_var_screeninfo *var = &fb_info->var;
-	struct fb_fix_screeninfo *fix = &fb_info->fix;
-
-	for (i = 0; i < h; i++) {
-		loc = (x + var->xoffset) * (var->bits_per_pixel / 8)
-			+ (y + i + var->yoffset) * fix->line_length;
-		memset(fbuffer + loc, 0, w * var->bits_per_pixel / 8);
-	}
-}
-
-static void fb_put_char(struct fb_info *fb_info, int x, int y, char c,
-		unsigned color)
-{
-	int i, j, bits, loc;
-	unsigned char *p8;
-	unsigned short *p16;
-	unsigned int *p32;
-	struct fb_var_screeninfo *var = &fb_info->var;
-	struct fb_fix_screeninfo *fix = &fb_info->fix;
-
-	unsigned short c16;
-	unsigned r = (color >> 16) & 0xff;
-	unsigned g = (color >> 8) & 0xff;
-	unsigned b = (color >> 0) & 0xff;
-
-	r = r * 32 / 256;
-	g = g * 64 / 256;
-	b = b * 32 / 256;
-
-	c16 = (r << 11) | (g << 5) | (b << 0);
-
-	for (i = 0; i < 8; i++) {
-		bits = fontdata_8x8[8 * c + i];
-		for (j = 0; j < 8; j++) {
-			loc = (x + j + var->xoffset) * (var->bits_per_pixel / 8)
-				+ (y + i + var->yoffset) * fix->line_length;
-			if (loc >= 0 && loc < (int)(fix->smem_len) &&
-					((bits >> (7 - j)) & 1)) {
-				switch (var->bits_per_pixel) {
-				case 8:
-					p8 =  fb_info->ptr + loc;
-					*p8 = color;
-				case 16:
-					p16 = fb_info->ptr + loc;
-					*p16 = c16;
-					break;
-				case 24:
-				case 32:
-					p32 = fb_info->ptr + loc;
-					*p32 = color;
-					break;
-				}
-			}
-		}
-	}
-}
-
-int fb_put_string(struct fb_info *fb_info, int x, int y, const char *s, int maxlen,
-		unsigned color, int clear, int clearlen)
-{
-	int i;
-	int w = 0;
-
-	if (clear)
-		fb_clear_area(fb_info, x, y, clearlen * 8, 8);
-
-	for (i = 0; i < (int)strlen(s) && i < maxlen; i++) {
-		fb_put_char(fb_info, (x + 8 * i), y, s[i], color);
-		w += 8;
-	}
-
-	return w;
-}
 /* end of fb-test.c */
 
-/*
- * font_8x8.c
- *
- * Author: Tomi Valkeinen <tomi.valkeinen@nokia.com>
- * Copyright (C) 2009-2012 Tomi Valkeinen
-
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of the GNU General Public License version 2 as published by
- * the Free Software Foundation.
- *
- * This program is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
- * more details.
- *
- * You should have received a copy of the GNU General Public License along with
- * this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
-/**********************************************/
-/*                                            */
-/*       Font file generated by cpi2fnt       */
-/*                                            */
-/**********************************************/
-
-char fontdata_8x8[] = {
```

Comment:
I really hate when data is allocated in .h files.

Reply:
I don't know it seemed obvious to me to include that in .h file.